collab-service/src/tunneller/collab_tunnel_redis.ts

```diff
@@ -0,0 +1,80 @@
+import { ServiceDefinition, UntypedServiceImplementation } from '@grpc/grpc-js';
+import { createClient, RedisClientType } from 'redis';
+import {
+  collabTunnelServiceDefinition,
+  ICollabTunnelService,
+} from '../proto/collab-service.grpc-server';
+import buildErrorResponse from '../controller/room_handler';
+import { createRedisPubSubAdapter } from '../redis_adapter/redis_pubsub_adapter';
+import createRoomSessionService from '../room_auth/room_session_agent';
+import loadEnvironment from '../utils/env_loader';
+import { Question } from '../proto/types';
+import getQuestionByDifficulty from '../controller/question_handler';
+
+const envConfig = loadEnvironment();
+
+const pub: RedisClientType = createClient({
+  url: envConfig.REDIS_SERVER_URL,
+});
+
+const sub: RedisClientType = createClient({
+  url: envConfig.REDIS_SERVER_URL,
+});
+
+pub.connect();
+sub.connect();
```

Comment:
I believe you should be able to use a single Redis client to connect for both (and in fact multiple) pub-sub topics.

Reply:
i did attempt the utilize a single client connection, but it is not possible according to Redis's documentation and after trying

https://redis.io/docs/manual/pubsub/
https://redis.io/commands/subscribe/
https://stackoverflow.com/questions/1505948/how-do-i-remove-a-single-file-from-the-staging-area-undo-git-add

Reply:
Ok, the other issue is dont initialize Redis clients as constants, it makes it impossible to test without a Redis server.

Reply:
is it possible to create a separate ticket/issue for this? converting the tunnel to not rely on the global constant into smth sort of like the ApiService has been a headache that i cant seem to figure out the various serviceImplementation, handlerDefinitions and Call framework and making a stream version or so. would appreciate your direct support.

Reply:
Fine with me